### PR TITLE
Fix No such function Error Occurs for Functions with Default Parameters

### PR DESCRIPTION
**Pull request by warunalakshitha**
## Purpose
Fixes https://github.com/ballerina-platform/ballerina-lang/issues/38171

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [x] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by gimantha — APPROVED** [review]
LGTM

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/Module.java [hunk 1]
```diff
@@ -36,19 +36,22 @@ public class Module {
     private String name;
     private String majorVersion;
     private int hashCode;
+    private boolean isTestPkg;
```
Comment (HindujaB) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/Module.java:39]: Shall we make this `final`? 
We can similarly change the other fields as well.

**Review by HindujaB — APPROVED** [review]
LGTM

**Review by Nadeeshan96 — APPROVED** [review]
LGTM

**Pull request merged**